### PR TITLE
Don't throw RangeError for toExponential(undefined)

### PR DESCRIPTION
**Pull request by ThomsonTan**
RangeError was raised for `Number.prototype.toExponential(undefined)` unexpectedly after refactoring range check (fraction should be between 1 to 20) out of else statement which converts non-integer and not undefined fractionDigits to integer. The spec also doesn't imply range error for it (see below link).

http://www.ecma-international.org/ecma-262/6.0/#sec-number.prototype.toexponential

### CODE REVIEW

File: lib/Runtime/Library/JavascriptNumber.cpp [hunk 1]
```diff
@@ -585,19 +585,23 @@ namespace Js
...
             //use the first arg as the fraction digits, ignore the rest.
             Var aFractionDigits = args[1];
+            bool noRangeCheck = false;
 
             // shortcut for tagged int's
             if(TaggedInt::Is(aFractionDigits))
             {
                 fractionDigits = TaggedInt::ToInt32(aFractionDigits);
             }
             else if(JavascriptOperators::GetTypeId(aFractionDigits) == TypeIds_Undefined)
```
Comment (akroshg) [inline lib/Runtime/Library/JavascriptNumber.cpp]: there is no need for this check, the ToInteger will convert it properly, right?

File: lib/Runtime/Library/JavascriptNumber.cpp [hunk 2]
```diff
@@ -585,19 +585,23 @@ namespace Js
...
-                ; // fraction digits = -1
+            {
```
Comment (akroshg) [inline lib/Runtime/Library/JavascriptNumber.cpp]: could you change this code to?
```js
            else if(JavascriptOperators::GetTypeId(aFractionDigits) != TypeIds_Undefined)
            {
                fractionDigits = (int)JavascriptConversion::ToInteger(aFractionDigits, scriptContext);
                if (fractionDigits < 0 || fractionDigits >20)
                {
                    JavascriptError::ThrowRangeError(scriptContext, JSERR_FractionOutOfRange);
                }
            }
        }


```

Comment (ThomsonTan) [inline lib/Runtime/Library/JavascriptNumber.cpp]: That will make the code behave a little different (like before the refactor). `fractionDigits` as `TaggedInt` will not be checked.

---
In reply to: [100205438](https://github.com/Microsoft/ChakraCore/pull/2497#discussion_r100205438) [](ancestors = 100205438)

Comment (akroshg) [inline lib/Runtime/Library/JavascriptNumber.cpp]: There was no point checking for taggedInt, the ToInteger is doing the same thing. Anyway - this is good to go.


---
In reply to: [100207392](https://github.com/Microsoft/ChakraCore/pull/2497#discussion_r100207392) [](ancestors = 100207392,100205438)

**Comment by akroshg** [comment]
:shipit:

**Comment by akroshg** [comment]
fixes #2428

**Pull request merged**